business/definitionService.js

```diff
@@ -135,29 +136,64 @@ class DefinitionService {
     const curation = await this.curationService.get(coordinates, curationSpec)
     const raw = await this.harvestService.getAll(coordinates)
     const summarized = await this.summaryService.summarizeAll(coordinates, raw)
-    const aggregated = await this.aggregationService.process(coordinates, summarized)
-    const definition = await this.curationService.apply(coordinates, curation, aggregated)
+    const tooledDefinition = (await this.aggregationService.process(coordinates, summarized)) || {}
+    this._ensureToolScores(coordinates, tooledDefinition)
+    const definition = await this.curationService.apply(coordinates, curation, tooledDefinition)
+    this._finalizeDefinition(coordinates, definition, curation)
+    this._ensureCuratedScores(definition)
+    return definition
+  }
+
+  // Compute and store the scored for the given definition but do it in a way that does not affect the
+  // definition so that further curations can be done.
+  _ensureToolScores(coordinates, definition) {
+    const rawDefinition = extend(true, {}, definition)
+    this._finalizeDefinition(coordinates, rawDefinition)
+    const { describedScore, licensedScore } = this._computeScores(rawDefinition)
+    set(definition, 'described.toolScore', describedScore)
+    set(definition, 'licensed.toolScore', licensedScore)
+  }
+
+  _ensureCuratedScores(definition) {
+    const { describedScore, licensedScore } = this._computeScores(definition)
+    set(definition, 'described.score', describedScore)
+    set(definition, 'licensed.score', licensedScore)
+  }
+
+  _finalizeDefinition(coordinates, definition, curation) {
     this._ensureFacets(definition)
     this._ensureCurationInfo(definition, curation)
     this._ensureSourceLocation(coordinates, definition)
     this._ensureCoordinates(coordinates, definition)
-    definition.score = this.computeScore(definition)
-    return definition
   }
 
-  /**
-   * Given a defintion, calculate a score for the definition
-   * @param {Defition} defintion
-   * @returns {number} The score for the definition
-   */
-  computeScore(definition) {
+  // Given a definition, calculate the scores for the definition and return an object with a score per dimension
+  _computeScores(definition) {
+    return {
+      licensedScore: this._computeLicensedScore(definition),
+      describedScore: this._computeDescribedScore(definition)
+    }
+  }
+
+  // Given a definition, calculate and return the score for the described dimension
+  _computeDescribedScore(definition) {
+    // @todo we need to flesh this out
+    // For now it just checks that a few props are present
+    let result = 0
+    result += !!get(definition, 'described.releaseDate')
+    result += !!get(definition, 'described.sourceLocation')
```

Comment:
might want to update to `result += !!get(definition, 'described.sourceLocation.url')`

no sense giving points if we can't actually get to the source code 😝